pandas/tests/test_tseries.py

```diff
@@ -337,6 +337,12 @@ def test_convert_infs():
     assert (result.dtype == np.float64)
 
 
+def test_scientific_no_exponent():
+    arr = np.array(['42E', '2E', '99e', '6e'], dtype='O')
```

Comment:
i don't see anything changed in the routines which support this, why is this test needed?